move pdb resource handling to the dynamic client

mrIncompetent opened the pull request:
**What this PR does / why we need it**:
Move PDB resource handling to the dynamic client.

**Release note**:
```release-note
NONE
```

mrIncompetent committed:
move pdb resource handling to the dynamic client

alvaroaleman commented on api/pkg/resources/apiserver/pdb.go:
```diff
@@ -8,26 +8,21 @@ import (
+func PodDisruptionBudgetCreator(data *resources.TemplateData) resources.PodDisruptionBudgetCreator {
+	return func(pdb *policyv1beta1.PodDisruptionBudget) (*policyv1beta1.PodDisruptionBudget, error) {
+		pdb.Name = resources.ApiserverPodDisruptionBudgetName
+		pdb.OwnerReferences = []metav1.OwnerReference{data.GetClusterRef()}
```
They are namespaced, right? We could just remove that

alvaroaleman commented on api/pkg/resources/etcd/pdb.go:
```diff
@@ -8,25 +8,20 @@ import (
+func PodDisruptionBudgetCreator(data *resources.TemplateData) resources.PodDisruptionBudgetCreator {
+	return func(pdb *policyv1beta1.PodDisruptionBudget) (*policyv1beta1.PodDisruptionBudget, error) {
+		pdb.Name = resources.EtcdPodDisruptionBudgetName
+		pdb.OwnerReferences = []metav1.OwnerReference{data.GetClusterRef()}
```
Required?

alvaroaleman commented on api/pkg/resources/metrics-server/pdb.go:
```diff
@@ -8,25 +8,20 @@ import (
+func PodDisruptionBudgetCreator(data *resources.TemplateData) resources.PodDisruptionBudgetCreator {
+	return func(pdb *policyv1beta1.PodDisruptionBudget) (*policyv1beta1.PodDisruptionBudget, error) {
+		pdb.Name = resources.MetricsServerPodDisruptionBudgetName
+		pdb.OwnerReferences = []metav1.OwnerReference{data.GetClusterRef()}
```
Required?

mrIncompetent committed:
remove owner refs from pdb's

mrIncompetent committed:
update fixtures

mrIncompetent committed:
add missing cluster ref

alvaroaleman commented:
/lgtm
/approve

mrIncompetent commented:
@alvaroaleman I removed the OwnerRefs inside the creator funcs with https://github.com/kubermatic/kubermatic/pull/2873/commits/8c92979ae77762e24fb14bebe24b569f3288be30.

I assumed we're attaching the OwnerRef via a `ObjectModifier` inside the `EnsureObject` function - which was only true from StatefulSets and Deployments. 
I added it for the missing objects in this PR as well: https://github.com/kubermatic/kubermatic/pull/2873/commits/eb0b23b0528be5e62c04964cafb69ea0d70c0e6c

The cluster controller still uses the OwnerRefs to trigger the cluster sync (Which will be replaced soon though)

alvaroaleman commented:
> The cluster controller still uses the OwnerRefs to trigger the cluster sync (Which will be replaced soon though)

Ah okay.  For openshift I have build a simple mapping func that just uses the namespace to determine the cluster to enq: https://github.com/kubermatic/kubermatic/blob/89c3ba3d4366b71905fab1066580e0141a5b2d7a/api/pkg/controller/openshift/openshift_controller.go#L55-L72

alvaroaleman commented:
/retest
